src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,37 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface TodoModalProps {
+  todo: Todo;
+  handleSetSelectedTodoId: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  handleSetSelectedTodoId,
+  todo,
+}) => {
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
You are initializing your state with props which is considered an anti-pattern in React. This is because the constructor or `useState` hook runs only once when the component is created. If the props change at some point in time, the state will not be updated with the new props. This can lead to bugs that are difficult to debug. You should consider using the `useEffect` hook to update the state when the props change.